src/std/foreign.ss

```diff
@@ -43,10 +44,172 @@
+                                        "if(" m " == NULL)" "\n"
+                                        m "= strdup(___arg2);" "\n"
+                                        "else if (strcmp(" m ", ___arg2) != 0) {" "\n"
+                                        "free(" m ");" "\n"
+                                        m "= strdup(___arg2);" "\n"
+                                        "}" "\n"
+                                        "___return;" "\n"))))
+               (default-free-body (and string-compat-required?
+                                       (string-append
+                                        "___SCMOBJ " struct-str "_ffi_free (void *ptr) {" "\n"
+                                        "struct " struct-str " *obj = (struct " struct-str "*) ptr;" "\n"
+                                        (apply string-append
+                                          (map (lambda (m)
+                                                 (case (cdr m)
+                                                   ((char-string)
+                                                    (let ((mem-name (symbol->string (car m))))
+                                                      (string-append "if(obj->" mem-name ") " 
+                                                                     "free(obj->" mem-name ");" "\n")))
+                                                                           (else "")))
+                                               members))
+                                        "free(obj);" "\n"
+                                        "return ___FIX (___NO_ERR);" "\n"
+                                        "}"
+                                        )))
+               (release-function (or release-function
+                                     (if string-compat-required?
+                                       (string-append struct-str "_ffi_free")
+                                       "ffi_free")))
+               (string-compat-types (if string-compat-required?
+                                      `((c-declare ,default-free-body)
+                                        (c-define-type ,shallow-ptr
+                                              (pointer ,struct (,struct-ptr) "ffi_free"))))))
+          `(begin
+             ,@(append `((c-define-type ,struct (struct ,struct-str))
+                         (c-define-type ,struct-ptr
+                                        (pointer ,struct (,struct-ptr) ,release-function))
+                         (c-define-type ,borrowed-ptr (pointer ,struct (,struct-ptr))))
+                       string-compat-types)
+
+
+                  (define ,(string->symbol (string-append struct-str "-ptr?"))
+                    (lambda (obj)
+                      (and (foreign? obj)
+                         (equal? (foreign-tags obj) (quote (,struct-ptr))))))
+
+                  ;; getter and setters
+                  ,@(apply append
```

Comment:
same here, we don't need the apply/append combo, simply splice them directly.

Reply:
here map is producing two elements in the list, so we would need a way to flatten them.

Reply:
ok for this, but the rest can be spliced in, no need to build a list just to append for that.